src/bootstrap/bin/main.rs

```diff
@@ -12,5 +12,31 @@ use bootstrap::{Build, Config};
 fn main() {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
+    check_version(&config);
     Build::new(config).build();
 }
+
+fn check_version(config: &Config) {
+    const VERSION: usize = 0;
+
+    let needs_warning = if let Some(version) = config.version {
+        if version != VERSION {
+            println!(
+                "warning: there have been changes to x.py since you last updated `version = {}`",
+                version
+            );
+            true
+        } else {
+            false
+        }
+    } else {
+        println!(
+            "warning: x.py has made several changes in the past month you may want to look at"
+        );
+        true
+    };
+    if needs_warning {
+        println!("help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`");
+        println!("note: to silence this warning, add `version = {}` to `config.toml`", VERSION);
+    }
```

Comment:
```suggestion
    if let Some(version) = config.version {
        if version != VERSION {
            println!("warning: x.py have new version but {} is used", version);
            println!("help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`");
            println!("note: to silence this warning, use `version = {}` in `config.toml`", VERSION);
        }
    } else {
        println!("warning: x.py have new changes you may want to look at");
        println!("help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`");
        println!("note: to silence this warning, add `version = {}` to `config.toml`", VERSION);
    }
```
Suggestion than I mentioned above.